docs/source/format/CProtocol.rst

```diff
@@ -0,0 +1,546 @@
++-----------------+---------------------------------------------------+------------+
+| ``tsn:...``     | timestamp [nanoseconds] with timezone "..."       | \(1)       |
++-----------------+---------------------------------------------------+------------+
+| ``tDs``         | duration [seconds]                                |            |
++-----------------+---------------------------------------------------+------------+
+| ``tDm``         | duration [milliseconds]                           |            |
++-----------------+---------------------------------------------------+------------+
+| ``tDu``         | duration [microseconds]                           |            |
++-----------------+---------------------------------------------------+------------+
+| ``tDn``         | duration [nanoseconds]                            |            |
++-----------------+---------------------------------------------------+------------+
+| ``tiM``         | interval [months]                                 |            |
++-----------------+---------------------------------------------------+------------+
+| ``tiD``         | interval [days, time]                             |            |
++-----------------+---------------------------------------------------+------------+
+
+Dictionary-encoded types do not have a specific format string.  Instead, the
+format string of the base array represents the dictionary index type, and the
+value type can be read from the dependent dictionary array (see below
+"Dictionary-encoded arrays").
+
+Nested types have multiple-character format strings starting with ``+``.  The
+names and types of child fields are read from the child arrays.
+
++------------------------+---------------------------------------------------+------------+
+| Format string          | Arrow data type                                   | Notes      |
++========================+===================================================+============+
+| ``+l``                 | list                                              |            |
++------------------------+---------------------------------------------------+------------+
+| ``+L``                 | large list                                        |            |
++------------------------+---------------------------------------------------+------------+
+| ``+w:123``             | fixed-sized list [123 items]                      |            |
++------------------------+---------------------------------------------------+------------+
+| ``+s``                 | struct                                            |            |
++------------------------+---------------------------------------------------+------------+
+| ``+m``                 | map                                               | \(2)       |
++------------------------+---------------------------------------------------+------------+
+| ``+u:I,J,...``         | union with type ids I,J...                        |            |
```

Comment:
we need to distinguish between sparse and dense?

Reply:
Oh, good catch.